mycroft/skills/skill_manager.py

```diff
@@ -115,20 +115,30 @@ def _shutdown_skill(instance):
 
 
 def _get_skill_locations(conf=None):
-    # next load all valid XDG paths
+    # the contents of each skills directory must be individual skill folders
+    # we are still dependent on the mycroft-core structure of skill_id/__init__.py
+
+    conf = conf or Configuration.get(remote=False)
+    # load all valid XDG paths
     # NOTE: skills are actually code, but treated as user data!
-    # they should be considered applets rather than applications
-    xdg_paths = list(reversed(
+    # they should be considered applets rather than full applications
+    skill_locations = list(reversed(
```

Comment:
Why is this reversed? Is the order determinate?

Reply:
the order is deterministic yes, we just happen to want the list ordered the reverse way of what the xdg module returns us. This was taken from the configuration logic i think